tests/testthat/test-implicit_assignment_linter.R

```diff
@@ -0,0 +1,103 @@
+test_that("implicit_assignment_linter skips allowed usages", {
+  linter <- implicit_assignment_linter()
+
+  expect_lint("x <- 1L", NULL, linter)
+  expect_lint("1L -> x", NULL, linter)
+  expect_lint("y <- if (is.null(x)) z else x", NULL, linter)
+
+  expect_lint("abc <- mean(1:4)", NULL, linter)
+  expect_lint("mean(1:4) -> abc", NULL, linter)
+
+  expect_lint(
+    trim_some("
+    x <- 1:4
+    mean(x)"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+    x <- 1L
+    if (x) TRUE"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+    0L -> abc
+    while (abc) {
+      FALSE
+    }"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+    if (x > 20L) {
+      x <- x / 2.0
+    }"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+    i <- 1
+    while (i < 6L) {
+      print(i)
+      i <- i + 1
+    }"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+    foo <- function(x) {
+      x <- x + 1
+      return(x)
+    }"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+    f <- function() {
+      p <- g()
+      p <- if (is.null(p)) x else p
+    }"),
+    NULL,
+    linter
+  )
```

Comment:
This test keeps failing. 🙈

![73mb2o](https://user-images.githubusercontent.com/11330453/206510094-995d3060-76ac-4685-b0eb-b9b33d70030d.jpg)